dataprofiler/data_readers/csv_data.py

```diff
@@ -534,13 +546,13 @@ def _load_data_from_str(self, data_as_str):
             )
         return data_utils.read_csv_df(
             data_buffered,
-            self.delimiter,
-            self.header,
+            cast(str, self.delimiter),
```

Comment:
do these need to be casted?

Reply:
Looks like with one of my recent commits to data_utils, `self.delimiter` doesn't need to be casted. However, `self.header` does because `read_csv_df` only supports `Optional[int]` but `self.header` could also be a string the `"auto"` case. As far as I can tell, this case doesn't exist at this point at runtime but it doesn't seem like mypy can detect that.